packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2371,15 +2395,23 @@ export class SnapController extends BaseController<
       const { iconPath } = manifest.result.source.location.npm;
       const svgIcon = iconPath ? await location.fetch(iconPath) : undefined;
 
-      const auxiliaryFiles = manifest.result.source.files
-        ? await Promise.all(
-            manifest.result.source.files.map(async (filePath) =>
-              location.fetch(filePath),
-            ),
-          )
-        : [];
+      const auxiliaryFiles = await getSnapFiles(
+        location,
+        manifest.result.source.files,
+      );
+
+      const localizationFiles = await getSnapFiles(
+        location,
+        manifest.result.source.locales,
+      );
 
-      const files = { manifest, sourceCode, svgIcon, auxiliaryFiles };
+      const files = {
+        manifest,
+        sourceCode,
+        svgIcon,
+        auxiliaryFiles,
+        localizationFiles: getValidatedLocalizationFiles(localizationFiles),
```

Comment:
Should we require somewhere that at least one localization file is found (maybe English) if the manifest is trying to use localization?

Reply:
We could. How should we detect if the manifest is trying to use localization? Check if there's any matches for the regex?

Reply:
That's one option yeah. I just wanna avoid a case where the snap proposes the name `{{name}}`